daily_report/report.py

```diff
@@ -0,0 +1,214 @@
+"""This file sends the daily report to users asynchronously."""
+import asyncio
+from datetime import datetime
+from os import environ as ENV
+
+import aioboto3
+import pandas as pd
+from boto3 import client
+from dotenv import load_dotenv
+from psycopg2 import connect
+
+
+WEATHER_EMOJIS = {
+    "Clear Sky": "☀️",
+    "Mainly Clear": "🌤",
+    "Partly Cloudy": "⛅",
+    "Overcast": "☁️",
+    "Fog": "🌫️",
+    "Depositing Rime Fog": "🌫️",
+    "Light Drizzle": "🌦️",
+    "Moderate Drizzle": "🌧️",
+    "Dense Drizzle": "🌧️",
+    "Light Freezing Drizzle": "🌧️",
+    "Dense Freezing Drizzle": "🌨️",
+    "Slight Rain": "🌦️",
+    "Moderate Rain": "🌧️",
+    "Heavy Rain": "🌧️",
+    "Light Freezing Rain": "🌧️",
+    "Heavy Freezing Rain": "🌨️",
+    "Slight Snowfall": "🌨️",
+    "Moderate Snowfall": "🌨️",
+    "Heavy Snowfall": "❄️",
+    "Snow grains": "❄️",
+    "Slight Rain Showers": "🌦️",
+    "Moderate Rain Showers": "🌦️",
+    "Violent Rain Showers": "🌧️",
+    "Slight Snow Showers": "🌨️",
+    "Heavy Snow Showers": "🌨️",
+    "Thunderstorm": "⛈️",
+    "Thunderstorm with Slight Hail": "⛈️",
+    "Thunderstorm with Heavy Hail": "⛈️"
+}
+
+
+def handler(event: dict = None, context: dict = None) -> None:
+    """This is designed to be compatible with the AWS Lambda function format."""
+    asyncio.run(main())
+
+
+async def main() -> None:
+    """Designed to asynchronously send emails."""
+    load_dotenv()
+    conn = get_db_connection(ENV)
+    df = prepare_data_frame(conn)
+    async with aioboto3.Session().client('ses', aws_access_key_id=ENV['AWS_KEY'],
+                                         aws_secret_access_key=ENV['AWS_SKEY'],
+                                         region_name='eu-west-2') as ses:
+        tasks = [asyncio.create_task(send_email(ses, await format_forecast_report(df, email), email))
+                 for email in df['email'].unique()]
+        await asyncio.gather(*tasks)
+    conn.close()
+
+
+def get_db_connection(config: dict) -> connect:
```

Comment:
Is this the right typehint? It returns a 'connection', not the 'connect' function itself. This comes from psycopg.extensions

Reply:
done